### PR TITLE
Allow list of chunks to be given to strip*()

### PR DESCRIPTION
**Pull request by michaelkschmidt**
This allows other BEAM languages to specify additional chunks that should be preserved.  The immediate need is 'Attr' for Elixir.

See https://github.com/whitfin/cachex/issues/205 for full details on the original issue.

### CODE REVIEW

**Comment by ferd** [comment]
Wait, does that mean that if I strip modules, I may end up with parts of my build keeping info I expect to be gone once in production? Some of these things you possibly do not want to ever hit a production server.

**Comment by michaelkschmidt** [comment]
@ferd I agree--the issue is that some Elixir modules (like cachex) need the 'Attr' chunk.  Without it `module.__info__(:attributes)`will not work.

Its standard practice for IoT devices to strip debug information prior to release.  This change only exposes an API to allow the list of chunks to be passed in.  It's still up to the end user to decide what chunks should be passed on to the final release.  Without this, you wind up having to do hacks like https://github.com/se-apc/strip_plugin/blob/master/lib/strip_plugin.ex

File: lib/stdlib/src/beam_lib.erl [hunk 1]
```diff
@@ -32,8 +32,12 @@
 	 all_chunks/1,
 	 diff_dirs/2,
 	 strip/1,
+	 strip/2,
 	 strip_files/1,
+	 strip_files/2,
 	 strip_release/1,
+	 strip_release/2,
+	 significant_chunks/0,
```
[the remark this answers is not in the document]
Comment (michaelkschmidt) [reply lib/stdlib/src/beam_lib.erl]: That is a good point.  I'll change it to prepend any additional chunks

**Comment by michaelkschmidt** [comment]
@jhogberg I made the suggested change and updated the docs.  Please let me know if you need anything else.

**Comment by jhogberg** [comment]
Thank you, can you add a test as well?

**Comment by michaelkschmidt** [comment]
@jhogberg sorry for the delay.  I believe I have added a test which verifies the additional chunks are preserved, and I squashed all commits into 1

**Comment by michaelkschmidt** [comment]
@jhogberg Are there any other changes you would like?

**Comment by jhogberg** [comment]
Sorry for the delay, I just got back from a cold. It looks good so I merged it to master, thanks again for the PR!

**Pull request merged**